cmd/api/src/api/v2/cypherquery.go

```diff
@@ -60,69 +59,69 @@ func (s Resources) CypherSearch(response http.ResponseWriter, request *http.Requ
 	}
 
 	if preparedQuery.HasMutation {
-		if !s.Authorizer.AllowsPermission(authCtx, auth.Permissions().GraphDBMutate) {
-			s.Authorizer.AuditLogUnauthorizedAccess(request)
-			api.WriteErrorResponse(
-				request.Context(),
-				api.BuildErrorResponse(http.StatusForbidden, "Permission denied: User may not modify the graph.", request), response,
-			)
-			return
-		}
-
-		// All mutation attempts must be audit logged even when failed
-		if auditLogEntry, err = model.NewAuditEntry(
-			model.AuditLogActionMutateGraph,
-			model.AuditLogStatusIntent,
-			model.AuditData{"query": preparedQuery.StrippedQuery}); err != nil {
-			api.WriteErrorResponse(
-				request.Context(),
-				api.BuildErrorResponse(http.StatusInternalServerError, api.ErrorResponseDetailsInternalServerError, request),
-				response,
-			)
-			return
-		}
+		graphResponse, err = s.cypherMutation(request, preparedQuery, payload.IncludeProperties)
+	} else {
+		graphResponse, err = s.GraphQuery.RawCypherQuery(request.Context(), preparedQuery, payload.IncludeProperties)
+	}
 
-		// create an intent audit log
-		if err := s.DB.AppendAuditLog(request.Context(), auditLogEntry); err != nil {
+	if err != nil {
+		switch true {
+		case strings.Contains(err.Error(), "Permission denied: User may not modify the graph."):
 			api.WriteErrorResponse(
 				request.Context(),
-				api.BuildErrorResponse(http.StatusInternalServerError, api.ErrorResponseDetailsInternalServerError, request),
-				response,
+				api.BuildErrorResponse(http.StatusForbidden, err.Error(), request), response,
 			)
-			return
-		}
-	}
-
-	if graphResponse, err := s.GraphQuery.RawCypherSearch(request.Context(), preparedQuery, payload.IncludeProperties); err != nil {
-		// audit log needs to be marked as failed if a mutation is present
-		if preparedQuery.HasMutation {
-			auditLogEntry.Status = model.AuditLogStatusFailure
-		}
-		if queries.IsQueryError(err) {
+		case queries.IsQueryError(err):
```

Comment:
This could probably be updated to `errors.As(err, queries.QueryError)` but I'm not 100% certain. Not a big deal, but would be nice to eventually kill this function

Reply:
I realized that we actually don't need the file whatsoever b/c we elevated the parse query function, it was the sole consumer of that error and those are all properly handled above